src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_Options.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal abstract partial class AbstractProject
+    {
+        private CommandLineArguments _lastParsedCommandLineArguments;
+
+        /// <summary>
+        /// Creates and sets new options using the last parsed command line arguments.
+        /// </summary>
+        protected void UpdateOptions()
+        {
+            Contract.ThrowIfNull(_lastParsedCommandLineArguments);
+
+            var newParseOptions = CreateParseOptions(_lastParsedCommandLineArguments);
+            var newCompilationOptions = CreateCompilationOptions(_lastParsedCommandLineArguments, newParseOptions);
+            if (newCompilationOptions == CurrentCompilationOptions && newParseOptions == CurrentParseOptions)
+            {
+                return;
+            }
+
+            SetOptions(newCompilationOptions, newParseOptions);
+        }
+
+        /// <summary>
+        /// Creates and sets new options using the given parsed command line arguments.
+        /// </summary>
+        protected void SetArgumentsAndUpdateOptions(CommandLineArguments commandLineArguments)
+        {
+            _lastParsedCommandLineArguments = commandLineArguments;
+            UpdateOptions();            
+        }
+
+        protected void SetOptions(CompilationOptions newCompilationOptions, ParseOptions newParseOptions)
```

Comment:
Looks like this is only used from UpdateOptions now above. Should this be either private or just inlined so that way nobody can call this and bypass the other code?